src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java

```diff
@@ -0,0 +1,169 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.junit.Test;
+import org.rulez.demokracia.pdengine.annotations.tested_behaviour;
+import org.rulez.demokracia.pdengine.annotations.tested_feature;
+import org.rulez.demokracia.pdengine.annotations.tested_operation;
+import org.rulez.demokracia.pdengine.exception.ReportedException;
+import org.rulez.demokracia.pdengine.testhelpers.CreatedDefaultVoteRegistry;
+
+public class VoteShowValidationTest extends CreatedDefaultVoteRegistry {
+
+	Vote vote;
+	String voteId;
+	String adminKey;
+	
+	public void setUp() throws ReportedException {
+		super.setUp();
+		vote = voteManager.getVote(adminInfo.voteId);
+		voteId = adminInfo.voteId;
+		adminKey = adminInfo.adminKey;
+	}
+
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("validates inputs")
+	@Test
+	public void invalid_voteId_is_rejected() throws ReportedException {
+		String invalidvoteId = RandomUtils.createRandomKey();
+		assertThrows(
+			() -> voteManager.showVote(invalidvoteId, adminInfo.adminKey)
+		).assertMessageIs(String.format("illegal voteId: %s",invalidvoteId));
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("validates inputs")
+	@Test
+	public void invalid_adminKey_is_rejected() throws ReportedException {
+		String invalidAdminKey = RandomUtils.createRandomKey();
+		assertThrows(
+			() -> voteManager.showVote(adminInfo.voteId, invalidAdminKey)
+		).assertMessageIs(String.format("Illegal adminKey: %s",invalidAdminKey));
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void check_name() throws ReportedException {
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("name"), vote.name);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void check_canAddIn() throws ReportedException {
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("canAddIn"), vote.canAddin);
+	}
+
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void check_creationTime() throws ReportedException {
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("creationTime"), vote.creationTime);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void check_choices() throws ReportedException {
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		JSONObject jsonObject = new JSONObject(vote.choices);
+
+		assertEquals(result.get("choices").toString(), jsonObject.toString());
```

Comment:
I am not convinced this properly serializes a non-empty set of choices.

Reply:
Ok I will check it.

Reply:
Ok I checked it, I have to create a toJson method for the Choices too.
The question is that do we need to test these methodes too?

Reply:
Fixed